OpenRA.Mods.Common/UtilityCommands/UpgradeRules.cs

```diff
@@ -2876,6 +2876,22 @@ internal static void UpgradeActorRules(int engineVersion, ref List<MiniYamlNode>
 					}
 				}
 
+				if (engineVersion < 20160109)
+				{
+					//Divide levels by 100
+					if (depth == 1 && (node.Key == "GainsExperience" || node.Key.StartsWith("GainsExperience@"))
+							&& node.Value.Nodes.FirstOrDefault(n => n.Key == "Upgrades") != null)
+					{
+						foreach (var kv in node.Value.Nodes.Where(n => n.Key == "Upgrades")) {
+							var key = float.Parse(kv.Key) / 100;							
+							// t is a MiniYamlNode that looks like ("200", "firepower, damage, speed, reload, inaccuracy, rank")
```

Comment:
Comments are for own benefit here, the code doesn't work yet. When I get it to work, the comments will be removed, and the commit rebased into the original one.